src/arch/riscv/arch.c

```diff
@@ -103,15 +103,17 @@ void arch_print_regs(struct nk_regs * r) {
 
 void * arch_read_sp(void) {
     void * sp = NULL;
-    __asm__ __volatile__ ( "mv %[_r], sp" : [_r] "=r" (rsp) : : "memory" );
+    __asm__ __volatile__ ( "mv %[_r], sp" : [_r] "=r" (sp) : : "memory" );
     return sp;
 }
 
 void arch_relax(void) {
-    asm volatile ("pause");
+    // RISCV HACK
+    // asm volatile ("pause");
 }
 
 void arch_halt(void) {
-    asm volatile ("hlt");
+    // RISCV HACK
+    // asm volatile ("hlt");
```

Comment:
In fact, riscv's wfi (which is what this should be) doesn't actually require the hardware wait. See the spec :)

Basically, wfi (hlt) is a recommendation to the hardware